modin/backends/pandas/parsers.py

```diff
@@ -153,15 +153,31 @@ def get_dtypes(cls, dtypes_ids):
             pandas.Series where index is columns names and values are
             columns dtypes.
         """
+        # each element in `partitions_dtypes` is a Series, where column names are
+        # used as index and types of columns for different partitions are used as values
         partitions_dtypes = cls.materialize(dtypes_ids)
         if all([len(dtype) == 0 for dtype in partitions_dtypes]):
             return None
-        return (
-            pandas.concat(partitions_dtypes, axis=1)
-            .apply(lambda row: find_common_type_cat(row.values), axis=1)
-            .squeeze(axis=0)
+
+        # concat all elements of `partitions_dtypes` and find common dtype
+        # for each of the column among all partitions
+        combined_dtypes = pandas.concat(partitions_dtypes, axis=1).apply(
+            lambda row: find_common_type_cat(row.values),
+            axis=1,
         )
 
+        for part_dtype in partitions_dtypes:
+            if not combined_dtypes.equals(part_dtype):
+                ErrorMessage.missmatch_with_pandas(
+                    operation="read_csv",
```

Comment:
Will it always be `read_csv` here? It could be the other readers, correct?

Reply:
Right, changed to `read_*`.